src/devices/cpu/cadr/cadr.cpp

```diff
@@ -0,0 +1,1007 @@
+{
+	if (BIT(m_ir, 25))
+	{
+		const u16 index = (m_ir >> 14) & 0x3ff;
+		m_a_mem[index] = output;
+		LOGMASKED(LOG_TRACE, "a[%x(%o)] <- %x(%o)\n", index, index, output, output);
+	}
+	else
+	{
+		switch ((m_ir >> 19) & 0x1f)
+		{
+		case 0x00: break;
+		case 0x01:
+			m_lc = (1 << 31) | (m_ic & 0x3c000000) | (output & 0x3ffffff);
+			if (!BIT(m_ic, 29))
+			{
+				m_lc &= ~1;
+			}
+			LOGMASKED(LOG_TRACE, "LC <- %x(%o)\n", output, output);
+			break;
+		case 0x02: // IC
+			m_ic = output;
+			m_lc = (m_lc & 0x83ffffff) | (m_ic & 0x3c000000);
+			LOGMASKED(LOG_INT, "sequence break %d\n", BIT(m_ic, 26));
+			LOGMASKED(LOG_INT, "interrupt enable %d\n", BIT(m_ic, 27));
+			LOGMASKED(LOG_INT, "bus reset %d\n", BIT(m_ic, 28));
+			LOGMASKED(LOG_INT, "lc mode %d\n", BIT(m_ic, 29));
+			break;
+		case 0x08: // PDL[ptr]
+			m_pdl[m_pdl_pointer] = output;
+			LOGMASKED(LOG_TRACE, "PDL[%x(%o)] <- %x(%o)\n", m_pdl_pointer, m_pdl_pointer, output, output);
+			break;
+		case 0x09: // PDL[ptr], push
+			m_pdl_pointer = (m_pdl_pointer + 1) & 0x3ff;
+			LOGMASKED(LOG_TRACE, "PDL[%x(%o)] <- %x(%o)\n", m_pdl_pointer, m_pdl_pointer, output, output);
+			m_pdl[m_pdl_pointer] = output;
+			break;
+		case 0x0a:
+			LOGMASKED(LOG_TRACE, "PDL[%x(%o)] <- %x(%o)\n", m_pdl_index, m_pdl_index, output, output);
+			m_pdl[m_pdl_index] = output;
+			break;
+		case 0x0b:
+			m_pdl_index = output & 0x3ff;
+			LOGMASKED(LOG_TRACE, "PDL-Index <- %x(%o)\n", m_pdl_index, m_pdl_index);
+			break;
+		case 0x0c:
+			m_pdl_pointer = output & 0x3ff;
+			LOGMASKED(LOG_TRACE, "PDL-ptr <- %x(%o)\n", m_pdl_pointer, m_pdl_pointer);
+			break;
+		case 0x0d: push_spc(output); break;
+		case 0x0e:
+			m_oa_reg_lo = output & 0x3ffffff;
+			LOGMASKED(LOG_TRACE, "OA-reg-lo <- %x(%o)\n", m_oa_reg_lo, m_oa_reg_lo);
+			break;
+		case 0x0f:
+			m_oa_reg_hi = output & 0x3fffff;
+			LOGMASKED(LOG_TRACE, "OA-reg-hi <- %x(%o)\n", m_oa_reg_hi, m_oa_reg_hi);
+			break;
+		case 0x10:
+			m_vma = output;
+			LOGMASKED(LOG_TRACE, "VMA <- %x(%o)\n", m_vma, m_vma);
+			break;
+		case 0x11: // VMA, start-read
+			m_vma = output;
+			read();
+			LOGMASKED(LOG_TRACE, "VMA <- %x(%o), start-read, MD=%x(%o)\n", m_vma, m_vma, m_md, m_md);
+			break;
+		case 0x12: // VMA, start-write
+			m_vma = output;
+			write();
+			LOGMASKED(LOG_TRACE, "VMA <- %x(%o), start-write, MD=%x(%o)\n", m_vma, m_vma, m_md, m_md);
+			break;
+		case 0x13: // VMA, write map
+			m_vma = output;
+			write_vma_map();
+			break;
+		case 0x18:
+			m_md = output;
+			LOGMASKED(LOG_TRACE, "MD <- %x(%o)\n", m_md, m_md);
+			break;
+		case 0x1a: // MD, start-write
+			m_md = output;
+			write();
+			LOGMASKED(LOG_TRACE, "MD <- %x(%o), start-write, MD=%x(%o)\n", m_md, m_md, m_md, m_md);
+			break;
+		case 0x1b: // MD, write map
+			m_md = output;
+			write_vma_map();
+			break;
+
+		case 0x19: // MD, start-read
+		default:
+		   fatalerror("%x(%o): output %02x not implemented", m_prev_pc, m_prev_pc, (m_ir >> 19) & 0x1f); break;
```

Comment:
There isn’t any point putting a `break` after a noreturn function (there are a few cases in this file).  If you want to make it more explicit that it’s leaving via an exception, you can change `fatalerror` to `throw emu_fatalerror`.